hq_superset/models.py

```diff
@@ -32,3 +45,71 @@ def update_dataset(self):
                 rows = list(cast_data_for_table(self.data, table))
                 insert_stmt = table.insert().values(rows)
                 connection.execute(insert_stmt)
+
+
+class OAuth2Client(db.Model, OAuth2ClientMixin):
+    __bind_key__ = OAUTH2_DATABASE_NAME
+    __tablename__ = 'hq_oauth_client'
+
+    domain = db.Column(db.String(255), primary_key=True)
+    client_secret = db.Column(db.String(255))  # more chars for encryption
+
+    def get_client_secret(self):
+        keys = get_fernet_keys()
+        fernet = MultiFernet(keys)
+
+        ciphertext_bytes = self.client_secret.encode('utf-8')
+        plaintext_bytes = fernet.decrypt(ciphertext_bytes)
+        return plaintext_bytes.decode('utf-8')
+
+    def set_client_secret(self, plaintext):
+        keys = get_fernet_keys()
+        fernet = MultiFernet(keys)
+
+        plaintext_bytes = plaintext.encode('utf-8')
+        ciphertext_bytes = fernet.encrypt(plaintext_bytes)
+        self.client_secret = ciphertext_bytes.decode('utf-8')
+
+    def check_client_secret(self, plaintext):
+        return self.get_client_secret() == plaintext
+
+    def revoke_tokens(self):
+        revoked_at = int(time.time())
+        stmt = (
+            update(OAuth2Token)
+            .where(OAuth2Token.client_id == self.client_id)
+            .where(OAuth2Token.access_token_revoked_at == 0)
+            .values(access_token_revoked_at=revoked_at)
```

Comment:
Can we add a docstring to this function please?
I assume it simply revokes all current active tokens for the client.

And since I don't understand this syntax well, just confirming that we don't need `values(OAuth2Token.access_token_revoked_at=revoked_at)`?